pkg/proxy/origins/clickhouse/model/model.go

```diff
@@ -0,0 +1,572 @@
+	2: marshalTimeseriesCSVWithNames,
+	3: marshalTimeseriesTSV,
+	4: marshalTimeseriesTSVWithNames,
+	5: marshalTimeseriesTSVWithNamesAndTypes,
+}
+
+// NewModeler returns a collection of modeling functions for influxdb interoperability
+func NewModeler() *timeseries.Modeler {
+	return &timeseries.Modeler{
+		WireUnmarshalerReader: UnmarshalTimeseriesReader,
+		WireMarshaler:         MarshalTimeseries,
+		WireMarshalWriter:     MarshalTimeseriesWriter,
+		WireUnmarshaler:       UnmarshalTimeseries,
+		CacheMarshaler:        dataset.MarshalDataSet,
+		CacheUnmarshaler:      dataset.UnmarshalDataSet,
+	}
+}
+
+func marshalTimeseriesJSON(ds *dataset.DataSet, rlo *timeseries.RequestOptions, w io.Writer) error {
+	return nil
+}
+
+func marshalTimeseriesCSV(ds *dataset.DataSet, rlo *timeseries.RequestOptions, w io.Writer) error {
+	return nil
+}
+
+func marshalTimeseriesCSVWithNames(ds *dataset.DataSet, rlo *timeseries.RequestOptions, w io.Writer) error {
+	return nil
+}
+
+func marshalTimeseriesTSV(ds *dataset.DataSet, rlo *timeseries.RequestOptions, w io.Writer) error {
+	return nil
+}
+
+func marshalTimeseriesTSVWithNames(ds *dataset.DataSet, rlo *timeseries.RequestOptions, w io.Writer) error {
+	return nil
+}
+
+func marshalTimeseriesTSVWithNamesAndTypes(ds *dataset.DataSet, rlo *timeseries.RequestOptions, w io.Writer) error {
+	return nil
+}
+
+// MarshalTimeseries converts a Timeseries into a JSON blob
+func MarshalTimeseries(ts timeseries.Timeseries, rlo *timeseries.RequestOptions) ([]byte, error) {
+	buf := bytes.NewBuffer(nil)
+	err := MarshalTimeseriesWriter(ts, rlo, buf)
+	return buf.Bytes(), err
+}
+
+// MarshalTimeseriesWriter converts a Timeseries into a JSON blob via an io.Writer
+func MarshalTimeseriesWriter(ts timeseries.Timeseries, rlo *timeseries.RequestOptions, w io.Writer) error {
+	if ts == nil {
+		return timeseries.ErrUnknownFormat
+	}
+	ds, ok := ts.(*dataset.DataSet)
+	if !ok {
+		return timeseries.ErrUnknownFormat
+	}
+	var of byte
+	if rlo != nil {
+		of = rlo.OutputFormat
+	}
+	marshaler, ok := marshalers[of]
+	if !ok {
+		return timeseries.ErrUnknownFormat
+	}
+	return marshaler(ds, rlo, w)
+}
+
+// UnmarshalTimeseries converts a JSON blob into a Timeseries
+func UnmarshalTimeseries(data []byte, trq *timeseries.TimeRangeQuery) (timeseries.Timeseries, error) {
+	buf := bytes.NewReader(data)
+	return UnmarshalTimeseriesReader(buf, trq)
+}
+
+// UnmarshalTimeseriesReader converts a JSON blob into a Timeseries via io.Reader
+func UnmarshalTimeseriesReader(reader io.Reader, trq *timeseries.TimeRangeQuery) (timeseries.Timeseries, error) {
+	return nil, nil
+}
+
+// /*
+// {"meta":[{"name":"t","type":"UInt64"},{"name":"cnt","type":"UInt64"}{"name":"meta1","type":"UInt16"},{"name":"meta2","type":"String"}]"data":[{"cnt":"12648509","meta1":200,"meta2":"value2","t":"1557766080000"},{"cnt":"10260032","meta1":200,"meta2":"value3","t":"1557766680000"},{"cnt":"1","meta1":206,"meta2":"value3","t":"1557767280000"}],"rows":3}
```

Comment:
Looks like a ton of ded code?

Reply:
while we wrote the marshalers for clickhouse (the part you noticed is commented out), the final common format interface meant they needed to be refactored. that appears to have been dropped on the floor hence the comments as a placeholder. will fix up in a subsequent PR prior to the first pre-release of 2.0